spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -302,6 +302,7 @@ protected void doInitialize() throws Exception {
 		if (this.taskScheduler == null) {
 			ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
 			threadPoolTaskScheduler.setThreadNamePrefix(getBeanName() + "-consumerMonitor-");
+			threadPoolTaskScheduler.setDaemon(true);
```

Comment:
H-m. Why don't call `ThreadPoolTaskScheduler.shutdown()` in the `actualShutDown()` instead?